### ISSUE TITLE
Creating and adding local audio track does not result in playout in local loopback

### ISSUE THREAD

**Issue by sebjf** [sebjf/webrtc#6]
Two peer connections in a Unity scene do not send one way audio using native devices as expected.

**Setup**: two peer connections with working signalling
**Action**: create local audio track at PC1
**Expected**: audio is received at PC2
**Actual**: no audio is received at PC2

Steps (in sequence):
1. Add local audio to PC1 : no audio
2. Add local audio to PC2 : PC2 -> PC1 only
3. Add local audio to PC1 : PC2 <-> PC1 duplex

It appears that audio frames are being processed by the mixer, but that they appear muted when received from the stream, though this may be an artefact of debugging.

**Comment by sebjf** [comment]
Placing an RTC_LOG call in audio_receive_stream.cc 284 

`  if (audio_frame->muted()) {
    RTC_LOG(LS_INFO) << "Audio frame muted";
  }`

shows that the received data stream is muted from PC1 to PC2, until the second call to create the local device and add the track.

**Comment by sebjf** [comment]
Looking at the neteq (neteq_impl.cc 788) it appears the packet buffer is empty. Placing a breakpoint in audio_send_stream.cc 397 confirms no stream transmission until local audio is added to pc2.

**Comment by sebjf** [comment]
SendAudioData is called by, at the top, DoCaptureThreadPollDMO() (audio_device_core_win.cc). This is not called until pc2 add. The thread is started by AudioDeviceWindowsCore::StartRecording() (audio_device_core_win.cc 2353).
This call fails (reporting error, which was hidden among info logs..)

```
        RTC_LOG(LS_ERROR)
            << "Playout must be started before recording when using"
            << " the built-in AEC";
```

This is because the _playing flag is false. This flag set in `AudioDeviceWindowsCore::StartPlayout()`, but for step (1) only on the second instance of the audio device (pc2's audio device).

StartPlayout is called by the familiar `void VoiceChannel::UpdateMediaSendRecvState_w()`, which is called when the receive stream is created. 

Our issue then looks like this:

**local audio track added to pc1**
a. pc1 fails to start capture because playout is false.
b. receive stream added to pc2 
c. pc2 playout activated by recieve stream

**local audio track added to pc2**
d. pc1 creates a receive stream, activating playout (but too late for original capture)
e. pc2 creates capture thread. playout is active so call succeeds.

**local audio track added to pc1 (again)**
f. pc1 creates capture thread, which succeeds because playout was activated by (d)

**Comment by sebjf** [comment]
The error is produced by this code block

```
    if (_builtInAecEnabled) {
      // Redirect to the DMO polling method.
      lpStartAddress = WSAPICaptureThreadPollDMO;

      if (!_playing) {
        // The DMO won't provide us captured output data unless we
        // give it render data to process.
        RTC_LOG(LS_ERROR)
            << "Playout must be started before recording when using"
            << " the built-in AEC";
        return -1;
      }
    }
```

I.e. the playout is only necessary for capture if the Windows native AEC is active. According to [this thread](https://groups.google.com/forum/#!topic/discuss-webrtc/G0UhzIF6-Jg), we don't want to use the in-built anyway because it's no good and support will likely be deprecated.

_builtInAecEnabled should be false by default though on the audio device.

It is enabled explicitly by the default AudioOptions struct of the VoiceEngine.

 ```
 if (options.echo_cancellation) {
    // Check if platform supports built-in EC. Currently only supported on
    // Android and in combination with Java based audio layer.
    // TODO(henrika): investigate possibility to support built-in EC also
    // in combination with Open SL ES audio.
    const bool built_in_aec = adm()->BuiltInAECIsAvailable();
    if (built_in_aec) {
      // Built-in EC exists on this device. Enable/Disable it according to the
      // echo_cancellation audio option.
      const bool enable_built_in_aec = *options.echo_cancellation;
      if (adm()->EnableBuiltInAEC(enable_built_in_aec) == 0 &&
          enable_built_in_aec) {
        // Disable internal software EC if built-in EC is enabled,
        // i.e., replace the software EC with the built-in EC.
        options.echo_cancellation = false;
        RTC_LOG(LS_INFO)
            << "Disabling EC since built-in EC will be used instead";
      }
    }
```
webrtc_voice_engine.cc, 379

The fix is to force the query of the Windows Core Audio Device to return null, by modifying this function:

```
bool AudioDeviceWindowsCore::BuiltInAECIsAvailable() const {
//  return _dmo != nullptr;
return false;
}
```
audio_device_core_win.cc, 2634

**Comment by sebjf** [comment]
Fixed in b19352dbaa1b9752f5fd5c25f6e351d0b10c55ad
